templates/deployment.yaml

```diff
@@ -54,46 +54,54 @@ spec:
           - /etc/docker/registry/config.yml
           ports:
             - containerPort: 5000
+            {{- if or .Values.metrics.enabled .Values.configData.http.debug.addr }}
+            - name: {{ .Values.metrics.targetPortName | default "debug" | quote }}
+              containerPort: {{ (split ":" .Values.configData.http.debug.addr)._1 }}
+            {{- else }}
+              {{- if .Values.metrics.serviceMonitor.enabled }}
+                {{- fail "You enabled the serviceMonitor without enabling the registry debug endpoint. Please set metrics.enabled: true or specify configData.http.debug.addr" }}
```

Comment:
"or" should be "and" here as well.

Basically, `metrics.enabled` only makes sense if the registry is configured to emit metrics, so it effectively depends on `configData.http.debug.addr` being set.